src/Gallery.CredentialExpiration/Job.cs

```
@@ -0,0 +1,198 @@
+
+                return false;
+            }
+            finally
+            {
+                // Make sure we know who has been contacted today, so they do not get double
+                // e-mail notifications.
+                File.WriteAllText(_cursorFile,
+                    JsonConvert.SerializeObject(_contactedUsersToday));
+            }
+
+            return true;
+        }
+
+        private async Task HandleExpiredCredentialEmail(ExpiredCredentialData expiredCredential)
+        {
+            _logger.LogInformation("Handling expired credential for user {Username} (expires: {Expires})...", expiredCredential.Username, expiredCredential.Expires);
+
+            // Build email body
+            string mailBody = null;
+            var expiresInDays = expiredCredential.Expires.UtcDateTime - DateTime.UtcNow;
+            if (expiresInDays.TotalDays <= 0)
+            {
+                mailBody = string.Format(Strings.ExpiredEmailBody, _galleryBrand, _galleryAccountUrl);
+            }
+            else
+            {
+                mailBody = string.Format(Strings.ExpiringEmailBody, _galleryBrand, _galleryAccountUrl, (int)expiresInDays.TotalDays);
+            }
+
+            // Build and send email
+            var mailMessage = new MailMessage(_mailFrom, expiredCredential.EmailAddress)
+            {
+                Subject = _mailSubject,
+                Body = mailBody
+            };
+
+            try
+            {
+                await _smtpClient.SendMailAsync(mailMessage);
+
+                _logger.LogInformation("Handled expired credential for user " + expiredCredential.Username + ".", expiredCredential.Username);
+
+                _contactedUsersToday.Add(expiredCredential.Username);
+            }
+            catch (Exception e)
+            {
+                _logger.LogCritical("Failed to handle expired credential for user {Username}.", expiredCredential.Username, e);
```

don't swallow the exception